src/utils/pgwire/src/pg_response.rs

```diff
@@ -118,6 +119,34 @@ where
 }
 
 impl StatementType {
+    pub fn infer_from_statement(stmt: &Statement) -> Result<Self, String> {
+        match stmt {
+            Statement::Query(_) => Ok(StatementType::SELECT),
+            Statement::Insert { returning, .. } => {
+                if returning.is_empty() {
+                    Ok(StatementType::INSERT)
+                } else {
+                    Ok(StatementType::INSERT_RETURNING)
+                }
+            }
+            Statement::Delete { returning, .. } => {
+                if returning.is_empty() {
+                    Ok(StatementType::DELETE)
+                } else {
+                    Ok(StatementType::DELETE_RETURNING)
+                }
+            }
+            Statement::Update { returning, .. } => {
+                if returning.is_empty() {
+                    Ok(StatementType::UPDATE)
+                } else {
+                    Ok(StatementType::UPDATE_RETURNING)
+                }
+            }
+            _ => Err("unsupported statement type".to_string()),
```

Comment:
Are these statement types sufficient  for us?

Reply:
It's sufficient now.🤣